File: guides/micronaut-data-r2dbc-repository/java/src/main/java/example/micronaut/GenreUpdateCommand.java
package example.micronaut;

import io.micronaut.serde.annotation.Serdeable;
import jakarta.validation.constraints.NotBlank;

@Serdeable // <1>
public class GenreUpdateCommand {
    private final long id;

    @NotBlank
    private final String name;

    public GenreUpdateCommand(long id, String name) {
        this.id = id;
        this.name = name;
    }

    public long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

}

File: guides/micronaut-data-jdbc-repository/python/src/example/micronaut/genre_update_command.py
from dataclasses import dataclass
from typing import Annotated

from jakarta.validation.constraints import NotBlank, NotNull
from micronaut.serde.annotation import Serdeable


@Serdeable  # <1>
@dataclass
class GenreUpdateCommand:
    id: Annotated[int, NotNull]
    name: Annotated[str, NotBlank]
